plugins/datadog/app/models/datadog_monitor.rb

```diff
@@ -44,28 +46,44 @@ def initialize(id, response = nil)
     @response = response
   end
 
-  def state
-    response[:overall_state]
+  # @return [String] nil, "Alert", "OK", "NoData"
+  # TODO: do we want to ignore muted / silenced scopes ?
+  def state(deploy_groups)
+    if match_source.present? && deploy_groups.presence
+      return nil unless groups = response.dig(:state, :groups)
+      deployed = deploy_groups.map { |dg| "#{match_target}:#{match_value(dg)}" }
+      alerting = groups.keys.flat_map { |g| g.to_s.split(",") }
+      (deployed & alerting).any? ? "Alert" : "OK"
+    else
+      response[:overall_state]
+    end
   end
 
   def name
     response[:name] || 'api error'
   end
 
   def url
-    "https://#{SUBDOMAIN}.datadoghq.com/monitors/#{@id}"
+    "#{BASE_URL}/monitors/#{@id}"
   end
 
   def reload
     @response = nil
   end
 
-  def alert?
-    state == "Alert"
-  end
-
   private
 
+  def match_value(deploy_group)
+    case match_source
+    when "deploy_group.permalink" then deploy_group.permalink
+    when "deploy_group.env_value" then deploy_group.env_value
+    when "environment.permalink" then deploy_group.environment.permalink
+    when "kubernetes_cluster.permalink"
```

Comment:
is kubernetes a plugin? do you need to guard if that's enabled?

Reply:
yes, but if someone does not have the kubernetes plugin and selects cluster matching then 💥 🤷‍♂